### ISSUE TITLE
paddle.nn.loss.CrossEntropyLoss()在GPU和CPU上计算误差较大

### ISSUE THREAD

**Issue by Aliceyins-dev** [PaddlePaddle/Paddle#48808]
### hard_label的交叉熵在GPU和CPU上计算误差大

```
import numpy as np
import paddle
import paddle.fluid as fluid

paddle.enable_static()
np.random.seed(33)
main_program = paddle.static.Program()
startup_program = paddle.static.Program()
main_program.random_seed = 33
startup_program.random_seed = 33

def test_softmax_with_cross_entropy():
    with paddle.utils.unique_name.guard():
        # soft_label=False  use_softmax=true
        with paddle.static.program_guard(main_program=main_program, startup_program=startup_program):
            data = paddle.static.data(name='data', shape=[64, 1000], dtype='float32')
            label = paddle.static.data(name='label', shape=[64], dtype='int32')
            data.stop_gradient = False
            label.stop_gradient = True
            loss = paddle.nn.loss.CrossEntropyLoss()
            out = loss(data, label)
            fetch_list = [out.name]
            g = paddle.static.gradients(out, [data, label])
            fetch_list.append(g[0].name)
            print(fetch_list)
            print(main_program)
            place = paddle.CUDAPlace(0)
            print("start to debug run")
            exe = paddle.static.Executor(place)
            # x = np.random.randn(48, 10).astype('float32')
            # y = np.random.randint(0, 10, (48)).astype('int32')
            x = np.load("./linear_5.tmp_1_gpu.npy")
            y = np.load("./reshape2_0.tmp_0_gpu.npy")
            output_gpu = exe.run(main_program, feed={"data": x, "label": y}, fetch_list=fetch_list, return_numpy=True)
            exec = paddle.static.Executor(paddle.CPUPlace())
            exec.run(startup_program)
            output_cpu = exec.run(main_program, feed={"data": x, "label": y}, fetch_list=fetch_list, return_numpy=True)
            print("output num:", len(output_gpu))
            for i in range(len(output_gpu)):
                print("------------")
                print(np.allclose(output_gpu[i], output_cpu[i], atol=1e-5, rtol=1e-5))
                print(fetch_list[i], output_gpu[i].reshape(-1))
                print(fetch_list[i], output_cpu[i].reshape(-1))
```
‘’‘
![image](https://user-images.githubusercontent.com/53935495/205923757-74e37ac4-8e0b-43c5-962d-849e23a67c4a.png)

’‘’

### 其他补充信息 Additional Supplementary Information
在某些数据分布下会出现此情况

可以将一些中间结果打印出来，跟数据分布有关系。

**Comment by Aliceyins-dev** [comment]
代码格式错乱了：
`import numpy as np
import paddle
import paddle.fluid as fluid

 
paddle.enable_static()
np.random.seed(33)
main_program = paddle.static.Program()
startup_program = paddle.static.Program()
main_program.random_seed = 33
startup_program.random_seed = 33


def test_softmax_with_cross_entropy():
    with paddle.utils.unique_name.guard():
        # soft_label=False  use_softmax=true
        with paddle.static.program_guard(main_program=main_program, startup_program=startup_program):
            data = paddle.static.data(name='data', shape=[64, 1000], dtype='float32')
            label = paddle.static.data(name='label', shape=[64], dtype='int32')
            data.stop_gradient = False
            label.stop_gradient = True
            loss = paddle.nn.loss.CrossEntropyLoss()
            out = loss(data, label)
            
            fetch_list = [out.name]
            g = paddle.static.gradients(out, [data, label])
            fetch_list.append(g[0].name)
            print(fetch_list)
            print(main_program)
            place = paddle.CUDAPlace(0)
            print("start to debug run")
            exe = paddle.static.Executor(place)
            # x = np.random.randn(48, 10).astype('float32')
            # y = np.random.randint(0, 10, (48)).astype('int32')
            x = np.load("./linear_5.tmp_1_gpu.npy")
            y = np.load("./reshape2_0.tmp_0_gpu.npy")
            output_gpu = exe.run(main_program, feed={"data": x, "label": y}, fetch_list=fetch_list, return_numpy=True)
            exec = paddle.static.Executor(paddle.CPUPlace())
            exec.run(startup_program)
            output_cpu = exec.run(main_program, feed={"data": x, "label": y}, fetch_list=fetch_list, return_numpy=True)
            print("output num:", len(output_gpu))
            for i in range(len(output_gpu)):
                print("------------")
                print(np.allclose(output_gpu[i], output_cpu[i], atol=1e-5, rtol=1e-5))
                print(fetch_list[i], output_gpu[i].reshape(-1))
                print(fetch_list[i], output_cpu[i].reshape(-1))`

**Comment by Shixiaowei02** [comment]
您好，问题已收到，我们正在尝试复现
